dropwizard-logging/src/main/java/io/dropwizard/logging/AbstractAppenderFactory.java

```diff
@@ -111,6 +114,8 @@
 
     private int discardingThreshold = -1;
 
+    private double maxMessagesPerSecond = -1;
```

Comment:
I can't help but think if someone set

```yaml
maxMessagesPerSecond: -2
```

that this should fail fast instead of ignoring it (eg. the dash was a typo). So I was thinking of using a validation constraint for the range (0, ∞). Unfortunately the `Min` annotation cannot represent an exclusive range, and `DecimalMin` isn't technically supported for `double` according to the bean validation spec -- but Hibernate Validator might support it

so you might want to try something like:

```java
    @DecimalMin(value = "0", inclusive = false)
    private Double maxMessagesPerSecond;
```

And write a test that ensures that validation fails if someone enters 0 or below. What do you think?

Reply:
Sounds good. It must be `@Nullable` as well, I guess.